distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso

```diff
@@ -672,26 +682,47 @@ type File
     list name_filter=Nothing recursive=False =
         all_files = case recursive of
             True -> here.list_descendants this
-            False -> here.list_immediate_children this
+            False -> this.list_immediate_children
         filtered = case name_filter of
             Nothing -> all_files
             _ ->
                 fs = FileSystems.getDefault
                 matcher = fs.getPathMatcher "glob:"+name_filter
                 all_files.filter file->
-                    pathStr = this.prim_file.relativize file.prim_file . getPath
+                    pathStr = this.relativize file . path
                     matcher.matches (Path.of pathStr)
         filtered
 
     ## UNSTABLE
 
        Checks if `this` is a child path of `other`.
-    is_child_of other = this.prim_file.startsWith other.prim_file
+    is_child_of : File -> Boolean
+    is_child_of other = this.starts_with other
 
     ## UNSTABLE
 
        Transforms `child` to a relative path with respect to `this`.
-    relativize child = File (this.prim_file.relativize child.prim_file)
+    relativize : File -> Boolean
+    relativize child = @Builtin_Method "File.relativize"
+
+
+    ## PRIVATE
+
+       Utility function that lists immediate children of a directory.
+    list_immediate_children : Vector.Vector File
+    list_immediate_children = Vector.Vector (this.list_immediate_children_array)
+
+    ## PRIVATE
+
+       Utility function that lists immediate children of a directory.
+    list_immediate_children_array : Array File
+    list_immediate_children_array = @Builtin_Method "File.list_immediate_children_array"
```

Comment:
Maybe instead we should call this with suffix (or prefix?) `_builtin`? Or `_primitive`? Then all 'internal' methods would have similar names which would be helpful understanding which are those.

Reply:
Good idea, but I'd rather leave it to a separate PR